xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGLES.cpp

```diff
@@ -53,65 +52,39 @@ CRenderBufferOpenGLES::CRenderBufferOpenGLES(AVPixelFormat format, AVPixelFormat
   m_width(width),
   m_height(height)
 {
-  size_t targetSize = m_width * m_height * glFormatElementByteCount(GL_RGBA); //! @todo: Get GLenum format from renderer
-  m_textureBuffer.resize(targetSize);
 }
 
 CRenderBufferOpenGLES::~CRenderBufferOpenGLES()
 {
   DeleteTexture();
-
-  if (m_swsContext != nullptr)
-    sws_freeContext(m_swsContext);
 }
 
 void CRenderBufferOpenGLES::CreateTexture()
 {
-  glEnable(m_textureTarget);
-
   glGenTextures(1, &m_textureId);
 
   glBindTexture(m_textureTarget, m_textureId);
 
-  glTexImage2D(m_textureTarget, 0, GL_LUMINANCE, m_width, m_height, 0, GL_LUMINANCE, GL_UNSIGNED_BYTE, NULL);
+  glTexImage2D(m_textureTarget, 0, GL_RGB, m_width, m_height, 0, GL_RGB, GL_UNSIGNED_SHORT_5_6_5, NULL);
 
   glTexParameteri(m_textureTarget, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
   glTexParameteri(m_textureTarget, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
   glTexParameteri(m_textureTarget, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
   glTexParameteri(m_textureTarget, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-
-  glDisable(m_textureTarget);
 }
 
 bool CRenderBufferOpenGLES::UploadTexture()
 {
-  if (m_textureBuffer.empty())
-  {
-    CLog::Log(LOGERROR, "Renderer: Unknown target texture size");
-    return false;
-  }
-
-  if (!CreateScalingContext())
-    return false;
-
   if (!glIsTexture(m_textureId))
     CreateTexture();
 
-  ScalePixels(m_data.data(), m_data.size(), m_textureBuffer.data(), m_textureBuffer.size());
-
-  glEnable(m_textureTarget);
-
-  const unsigned int bpp = 1;
-  glPixelStorei(GL_UNPACK_ALIGNMENT, bpp);
-
-  const unsigned datatype = GL_UNSIGNED_BYTE;
-
   glBindTexture(m_textureTarget, m_textureId);
-  glTexSubImage2D(m_textureTarget, 0, 0, 0, m_width, m_height, GL_LUMINANCE, datatype, m_textureBuffer.data());
 
-  glBindTexture(m_textureTarget, 0);
+  int stride = GetFrameSize() / m_height;
```

Comment:
as a matter of taste, since delving into Rust, I prefer

```c++
const int stride = GetFrameSize() / m_height;
```

when I know the variable's not going to change